src/plugins/data/server/search/opensearch_search/opensearch_search_strategy.ts

```diff
@@ -73,7 +74,19 @@ export const opensearchSearchStrategyProvider = (
       });
 
       try {
-        const client = await decideClient(context, request, withLongNumeralsSupport);
+        const isOpenSearchHostsEmpty =
+          openSearchServiceSetup?.legacy.client.config.hosts.length === 0;
+
+        if (dataSource?.dataSourceEnabled() && isOpenSearchHostsEmpty && !request.dataSourceId) {
+          throw new Error(`Data source id is required when no openseach hosts config provided`);
+        }
+
+        const client = await decideClient(
+          context,
+          request,
+          dataSource?.dataSourceEnabled(),
```

Comment:
Let's double check if we really need this? Can we use context?

Reply:
Already reply in last comment. 
This one is not needed and removed. Thanks~